src/analysis/inner_product_space/projection.lean

```diff
@@ -857,19 +857,24 @@ lemma id_eq_sum_orthogonal_projection_self_orthogonal_complement
   + Kᗮ.subtypeL.comp (orthogonal_projection Kᗮ) :=
 by { ext w, exact eq_sum_orthogonal_projection_self_orthogonal_complement K w }
 
+@[simp] lemma inner_orthogonal_projection_eq_of_mem_right [complete_space K] (u : K) (v : E) :
+  ⟪orthogonal_projection K v, u⟫ = ⟪v, u⟫ :=
+calc ⟪orthogonal_projection K v, u⟫
+    = ⟪(orthogonal_projection K v : E), u⟫ : K.coe_inner _ _
+... = ⟪(orthogonal_projection K v : E), u⟫ + ⟪v - orthogonal_projection K v, u⟫ :
+      by rw [orthogonal_projection_inner_eq_zero _ _ (submodule.coe_mem _), add_zero]
+... = ⟪v, u⟫ :
+      by rw [← inner_add_left, add_sub_cancel'_right]
+
+@[simp] lemma inner_orthogonal_projection_eq_of_mem_left [complete_space K] (u : K) (v : E) :
+  ⟪u, orthogonal_projection K v⟫ = ⟪(u : E), v⟫ :=
+by rw [← inner_conj_sym, ← inner_conj_sym (u : E), inner_orthogonal_projection_eq_of_mem_right]
+
 /-- The orthogonal projection is self-adjoint. -/
-lemma inner_orthogonal_projection_left_eq_right [complete_space E]
+lemma inner_orthogonal_projection_left_eq_right
```

Comment:
While you're at it, could you please change the statement and lemma name to be in terms of `is_self_adjoint`?

Reply:
The problem is that `inner_product_space.is_self_adjoint` is defined in `analysis/inner_product_space/adjoint`, which imports this. Should I remove this version completely and only keep the already existing https://leanprover-community.github.io/mathlib_docs/analysis/inner_product_space/adjoint.html#inner_product_space.orthogonal_projection_is_self_adjoint or should I keep both ?

Reply:
For me, this is a reason to keep `inner_product_space.is_self_adjoint` (soon to be renamed `is_symmetric`) earlier in the import hierarchy, undoing the move of #15281.  Let's see what @mcdoll thinks.

Reply:
We could also make a separate small file for the theory of operators that are `inner_product_space.is_self_adjoint`-(soon-to-be-renamed-`is_symmetric`).

Reply:
Yes I'd be in favor of having the new `is_self_adjoint` where the current one is but moving `is_symmetric` earlier. Shall we put this PR on hold then ?

Reply:
Sure, hopefully only for a day or two while we discuss with Moritz and other interested parties!

Reply:
I think that having a small file for symmetric operators would be better - I think having `is_symmetric` in `basic` is not a good idea since the file is way too large anyways.
Another random remark: golfing `complete_space` away is nice from the mathlib point of view, but mathematically completely irrelevant since nobody (at least I don't know anyone) cares about non-complete spaces - you always just take the completion and are happy.

Reply:
I'm not claiming this is super interesting mathematically. That said, taking the orthogonal projection of a polynomial onto a finite dimensional subset can be useful, right ? Or maybe I'm making things up, I'm not rue :sweat_smile:

Reply:
And imo it makes the proof cleaner anyway

Reply:
I am sorry for not being clear, I think this PR is good.

Reply:
Oh I didn't take it as being against the PR, don't worry, I just wanted to give my thoughts on the matter.